distiller/extraction/analyze.py

```diff
@@ -106,18 +106,39 @@ def output_as_csv(results):
     parser.add_argument("filename")
     parser.add_argument("--csv", help="store output to a CSV file")
     parser.add_argument("--pickle", help="store output to a pickle file")
+    parser.add_argument("--errors", help="record errors", action="store_true")
     args = parser.parse_args()
 
     pdf = files.input_file(args.filename, mode="rb")
-    errors = pdf_utils.errors(pdf)
-    if errors:
-        print(f"Could not read file {args.filename}: {errors}. Bailing out.")
+    pdf_errors = pdf_utils.errors(pdf)
+
+    if pdf_errors and args.errors:
+        if args.pickle:
+            with files.output_file(args.pickle, mode="wb") as fd:
+                pickle.dump(pdf_errors, fd)
+        elif args.csv:
+            with files.output_file(args.csv, mode="w") as fd:
+                cw = csv.writer(fd)
+                cw.writerow(["error"])
+                for line in pdf_errors:
+                    cw.writerow([line])
+        else:
+            for error in errors:
+                print(error)
+        sys.exit(1)
+
+    if pdf_errors:
+        print(f"Could not read file {args.filename}: {pdf_errors}. Bailing out.")
         sys.exit(1)
 
     print(f"processing {args.filename}")
     processor = nlp.setup()
     audit_results = analyze(processor, pdf, silent=True)
 
+    if not audit_results:
+        print(f"processed {args.filename} with no results")
+        sys.exit(1)
```

Comment:
Small thing - but if we had an error code returned from this function for the `sys.exit(1)` cases (or an exception raised), you could handle that error in the management command and raise `CommandError` - https://docs.djangoproject.com/en/3.0/howto/custom-management-commands/#django.core.management.CommandError.

Reply:
Good point -- if we were running a Django command, we'd def want to return a value and raise `CommandError`. But here, we're running as a module (`__name__ == "__main__"`) so we have to do `sys.exit`. Ideally this module would be run locally for bulk updates (I shudder to think how long it would take cloud.gov to process 250k+ pdf files), and the django command for new incoming pdfs.